resources/ec2_security_groups.go

```diff
@@ -402,83 +235,74 @@ func resolveEc2securityGroupTags(ctx context.Context, meta schema.ClientMeta, re
 	return resource.Set("tags", tags)
 }
 func fetchEc2SecurityGroupIpPermissions(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+
 	securityGroup, ok := parent.Item.(types.SecurityGroup)
 	if !ok {
 		return fmt.Errorf("not ec2 security group")
 	}
-	res <- securityGroup.IpPermissions
-	return nil
-}
-func fetchEc2SecurityGroupIpPermissionIpRanges(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
-	securityGroupIpPermission, ok := parent.Item.(types.IpPermission)
-	if !ok {
-		return fmt.Errorf("not ec2 security group ip permission")
+
+	capacity := len(securityGroup.IpPermissionsEgress) + len(securityGroup.IpPermissions)
+
+	ipRanges := make([]ipPermission, 0, capacity)
+	for _, ip := range securityGroup.IpPermissionsEgress {
+		ipRanges = append(ipRanges, ipPermission{ip, "egress"})
 	}
-	res <- securityGroupIpPermission.IpRanges
-	return nil
-}
-func fetchEc2SecurityGroupIpPermissionIpv6Ranges(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
-	securityGroupIpPermission, ok := parent.Item.(types.IpPermission)
-	if !ok {
-		return fmt.Errorf("not ec2 security group ip permission")
+
+	for _, ip := range securityGroup.IpPermissions {
+		ipRanges = append(ipRanges, ipPermission{ip, "ingress"})
 	}
-	res <- securityGroupIpPermission.Ipv6Ranges
+	res <- ipRanges
 	return nil
 }
-func fetchEc2SecurityGroupIpPermissionPrefixListIds(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
-	securityGroupIpPermission, ok := parent.Item.(types.IpPermission)
-	if !ok {
-		return fmt.Errorf("not ec2 security group ip permission")
+
+func getString(s *string) string {
```

Comment:
aws sdk already has a fn for that, `aws.String(*string) string`

Reply:
Thank you! it is `aws.ToString`